CHANGELOG.mkd

```diff
@@ -4,6 +4,7 @@ CHANGELOG
 Unreleased
 ----------
 
+- (RK-381) Do not purge files that are pointed to by a symlink. [#1233](https://github.com/puppetlabs/r10k/pull/1233)
```

Comment:
Would it be clearer to say something like "Do not follow symlinks when purging files."? Or maybe "Do not purge symlink target files."? Just thinking "files that are pointed to by a symlink" sounds a little clunky. ¯\_(ツ)_/¯

Reply:
Yeah I can rephrase. 

I did it like this because I find the terminology around symlinks to be very confusing: people avoid the word "target" and call it the "source" file instead, which has always seemed a little backwards to me, because I think we have a tendency to think of an arrow with a source (the location of the link) and a target (the file pointed to), but that's the exact opposite of how Unix talks about them.

And I didn't say "follow symlinks when purging" because that's not really what was happening (and I spent way too long thinking that it was). We previously we were recursing into symlinked directories _explicitly_ when figuring out which files to purge. But that subtlety probably isn't important here.

Reply:
heh yeah, I do know what you mean about symlink terminology being pretty confusing. I guess in my mind "source" and "target" both mean the real file, as opposed to the symlink pointing to it haha. I can definitely see how "target" could mean either, but do you think "source" is clear?

Reply:
You could also just copy the title of this PR ¯\_(ツ)_/¯